src/operator/contrib/transformer-inl.h

```diff
@@ -61,6 +61,229 @@ static void DivSqrtDimForward_(const nnvm::NodeAttrs& attrs,
   });
 }
 
+
+
+struct SldWinAttenParam : public dmlc::Parameter<SldWinAttenParam> {
+  int w;
+  bool symmetric;
+  DMLC_DECLARE_PARAMETER(SldWinAttenParam) {
+    DMLC_DECLARE_FIELD(w)
+    .describe("The one-sided window length");
+    DMLC_DECLARE_FIELD(symmetric)
+    .describe("Whether to use causal attention");
```

Comment:
Either `symmetric` or `causal` looks good. If we keep it as `symmetric`, we may mention that `when symmetric is False, it will use the one-sided causal attention`.